src/coreclr/gc/handletablescan.cpp

```diff
@@ -271,6 +271,17 @@ struct ScanQNode
  */
 #define INCLUSION_MAP_SIZE (HANDLE_MAX_INTERNAL_TYPES + 1)
 
+/**
+ * WhichGeneration
+ *
+ * Determine the generation of an object (in which frozen objects are identified as Gen2)
+ */
+int WhichGeneration(_UNCHECKED_OBJECTREF * obj)
+{
+    int generation = g_theGCHeap->WhichGeneration(obj);
+    return generation == g_theGCHeap->GetMaxGeneration() : generation;
```

Comment:
(I do not think this is going to compile. It is missing `?`.)

`GetMaxGeneration` is a virtual call. Should we use `max_generation` constant instead? It should be accessible in this file already.